libethereum/Account.h

```diff
@@ -161,7 +165,7 @@ class Account
 	h256 codeHash() const { assert(!isFreshCode()); return m_codeHash; }
 
 	/// Sets the code of the account. Must only be called when isFreshCode() returns true.
-	void setCode(bytes&& _code) { assert(isFreshCode()); m_codeCache = std::move(_code); changed(); }
+	void setCode(bytes&& _code) { m_codeCache = std::move(_code); changed(); }
```

Comment:
In general, I'm using it to also clear the code when reverting. There are alternatives to this, but I believe this API is too protective.